### PR TITLE
Upgrade line chart to amCharts 5

### PR DESCRIPTION
**Pull request by NXXR**
### Description
This PR updates the line chart to use amCharts 5.

#### Related Issues
- #68 
- #93 

### Checklist

**I, the author of this PR checked the following requirements for good software quality:**

- [x] The code is properly formatted (I ran the formatter)
- [x] The code is written with our software quality standards (I ran the linter)
- [x] The code is written using our code style
- [x] Extensive in source documentation has been added
- [x] ~~Unit and/or integration tests have been added~~
- [x] All texts have been internationalized with at least the following languages:
  - [x] English
  - [x] German
- [x] I tried addressing all new accessibility problems displayed in the console and documented if they can't be fixed
- [x] I attached performance measurements to prevent performance degradation
- [x] I added the changes to the next release section of the [changelog](../frontend/docs/changelog/changelog-en.md)

**I, the reviewer checked the following things:**

- [x] I ran the software once and tried all new and related functionality to this PR
- [x] I looked at all new and changed lines of code and commented on possible problems
- [x] I read the added documentation and checked if it is understandable and clear
- [x] I checked the added tests for completeness
- [x] I checked the internationalized strings for spelling errors
- [x] I checked the performance metrics for problems or unexplained degradation
- [x] I checked that the changes are noted in the [changelog](../frontend/docs/changelog/changelog-en.md)

### CODE REVIEW

File: frontend/src/components/SimulationChart.tsx [hunk 1]
```diff
@@ -93,358 +97,565 @@ export default function SimulationChart(): JSX.Element {
...
+            {timeUnit: 'month', count: 1},
+            {timeUnit: 'month', count: 3},
+            {timeUnit: 'year', count: 1},
+          ],
+          // Add tooltip instance so cursor can display value
+          tooltip: Tooltip.new(root, {}),
+        })
+      );
+      // Change axis renderer to have ticks/labels on day center
+      const Xrenderer = xAxis.get('renderer');
```
Comment (JonasGilg) [inline frontend/src/components/SimulationChart.tsx:145]:
```suggestion
      const xRenderer = xAxis.get('renderer');
```

File: frontend/src/components/SimulationChart.tsx [hunk 2]
```diff
@@ -93,358 +97,565 @@ export default function SimulationChart(): JSX.Element {
...
+      Xrenderer.ticks.template.setAll({
```
Comment (JonasGilg) [inline frontend/src/components/SimulationChart.tsx:146]:
```suggestion
      xRenderer.ticks.template.setAll({
```

**Review by JonasGilg — CHANGES_REQUESTED** [review]

**Comment by JonasGilg** [comment]
Can you make the date selection a single click again?

**Comment by NXXR** [comment]
> Can you make the date selection a single click again?

sorry,
forgot yesterday.
is fixed now.

**Review by JonasGilg — DISMISSED** [review]
Feels much better now. Sadly I discovered another issue. If you toggle a scenario or group filters the time selector goes missing :(

**Comment by NXXR** [comment]
> Feels much better now. Sadly I discovered another issue. If you toggle a scenario or group filters the time selector goes missing :(

looking into it.
It seems to only happen when you activate/deactivate the first scenario or group filters, but not with the second scenario.
Can you confirm that?

**Comment by JonasGilg** [comment]
> It seems to only happen when you activate/deactivate the first scenario or group filters, but not with the second scenario.
Can you confirm that?

Also happens with other scenarios/filters and when reloading the page.

**Comment by NXXR** [comment]
guide dissapperaing (#252 ) should be fixed now

File: frontend/src/components/SimulationChart.tsx [hunk 3]
```diff
@@ -174,6 +165,64 @@ export default function SimulationChart(): JSX.Element {
...
+    },
+    // This effect should only run once. dispatch should not change during runtime
+    [dispatch]
+  );
+
+  // Effect to change localization of chart if language changes
+  useEffect(
+    () => {
+      // Skip if root or chart is not initialized
+      if (!rootRef.current || !chartRef.current) return;
```
Comment (JonasGilg) [inline frontend/src/components/SimulationChart.tsx:199]:
```suggestion
      if (!rootRef.current || !chartRef.current) {
        return;
      }
```

File: frontend/src/components/SimulationChart.tsx [hunk 4]
```diff
@@ -174,6 +165,64 @@ export default function SimulationChart(): JSX.Element {
...
+
+      // Set refs to be used in other effects
+      rootRef.current = root;
+      chartRef.current = chart;
+
+      // Clean-up before re-running this effect
+      return () => {
+        // Dispose old root and chart before creating a new instance
+        chartRef.current && chartRef.current.dispose();
+        rootRef.current && rootRef.current.dispose();
```
Comment (JonasGilg) [inline frontend/src/components/SimulationChart.tsx:188]:
```suggestion
        chartRef.current?.dispose();
        rootRef.current?.dispose();
```

File: frontend/src/components/SimulationChart.tsx [hunk 5]
```diff
@@ -174,6 +165,64 @@ export default function SimulationChart(): JSX.Element {
...
+    },
+    // Re-run effect if language changes
+    [i18n.language, t]
+  );
+
+  // Effect to add series to chart
+  useEffect(
+    () => {
+      // Skip if root or chart not initialized
+      if (!rootRef.current || !chartRef.current) return;
```
Comment (JonasGilg) [inline frontend/src/components/SimulationChart.tsx:219]:
```suggestion
      if (!rootRef.current || !chartRef.current) {
        return;
      }
```

File: frontend/src/components/SimulationChart.tsx [hunk 6]
```diff
@@ -354,8 +386,8 @@ export default function SimulationChart(): JSX.Element {
   // Effect to add Guide when date selected
   useEffect(
     () => {
-      // Skip effect if chart is not initialized yet or no date is selected
-      if (!chartRef.current || !selectedDate) return;
+      // Skip effect if chart (or root) is not initialized yet or no date is selected
+      if (!chartRef.current || !rootRef.current || !selectedDate) return;
```
Comment (JonasGilg) [inline frontend/src/components/SimulationChart.tsx:390]:
```suggestion
      if (!chartRef.current || !rootRef.current || !selectedDate) {
        return;
      }
```

**Review by JonasGilg — CHANGES_REQUESTED** [review]

**Review by JonasGilg — APPROVED** [review]

**Pull request merged**